replication/binlogsyncer.go

```diff
@@ -169,8 +170,14 @@ func NewBinlogSyncer(cfg BinlogSyncerConfig) *BinlogSyncer {
 
 	// Clear the Password to avoid outputing it in log.
 	pass := cfg.Password
-	cfg.Password = ""
-	cfg.Logger.Infof("create BinlogSyncer with config %v", cfg)
+	cfg.Password = "<hidden>"
+	jsonCfg, err := json.Marshal(cfg)
```

Comment:
ping @dveeden CI found some problems

Reply:
I can suggest using `%+v` instead `%v` or json.

And instead something like
`{100 mysql 127.0.0.1 3306 root secret   false false <nil> false UTC false 0 0s 0s 0 false false 0 <nil> <nil> <nil> <nil> false}`
we will get
`{ServerID:100 Flavor:mysql Host:127.0.0.1 Port:3306 User:root Password:secret Localhost: Charset: SemiSyncEnabled:false RawModeEnabled:false TLSConfig:<nil> ParseTime:false TimestampStringLocation:UTC UseDecimal:false RecvBufferSize:0 HeartbeatPeriod:0s ReadTimeout:0s MaxReconnectAttempts:0 DisableRetrySync:false VerifyChecksum:false DumpCommandFlag:0 Option:<nil> Logger:<nil> Dialer:<nil> RowsEventDecodeFunc:<nil> DiscardGTIDSet:false}`

Reply:
I think `%+v` is a very good option. It makes the output clear and readable without adding any complexity. Do any others have an opinion on this?

Reply:
lgtm